ultravox/training/train.py

```diff
@@ -129,10 +118,18 @@ def train(args: config_base.TrainConfig):
 
     logging.info("Instantiating model...")
 
-    # Since the model downloads the language model and audio encoder weights, we want one process to finish up
-    # downloading before the others start in order to avoid race conditions.
-    with ddp_utils.run_on_master_first(is_master):
-        model = ultravox_model.UltravoxModel(config)
+    if is_distributed:
```

Comment:
Why do this only when we're running on multi-GPU?

Reply:
Short answer: on a single process there is no race condition, so there's no reason to force that `prefetch_weights.py` was run.

TBH I'm conflicted about whether we even need to enforce this, why raise an error only in anticipation of something going wrong. I added it because @zqhuang211 asked for it.

But the idea is that there are two scenarios:
1. if there's only one process, running `train.py` without `prefetch_weights.py` is fine. No reason to panic.
2. if there are multiple processes, we are likely to get an error, so we'll have to run prefetch_weights. The point of this code is to make the error message more easily understandable I guess

Reply:
A very minor nit: it's possible to have multiple GPUs and still have one process (as in DataParallel).